subsys/net/ip/tcp.c

```diff
@@ -997,6 +1020,11 @@ static int tcp_unsent_len(struct tcp *conn)
 	}
 
 	unsent_len = conn->send_data_total - conn->unacked_len;
+	if (conn->unacked_len >= conn->send_win) {
+		unsent_len = 0;
```

Comment:
When could this happen? Peer reduced RX window size on his side?

Reply:
Indeed, although unlikely it could happen.